### PR TITLE
Eip saga

### PR DESCRIPTION
**Pull request by Delawen**

### CODE REVIEW

File: kamelet-support/src/main/java/io/kaoto/backend/model/deployment/kamelet/step/ToFlowStep.java [hunk 1]
```diff
@@ -12,23 +12,34 @@
...
+            UriFlowStep uri = new UriFlowStep();
+            uri.setUri(sto);
+            setTo(uri);
+        } else if (to instanceof Map map) {
+            UriFlowStep uri = new UriFlowStep();
+            uri.setUri(map.getOrDefault("uri", "").toString());
+            var parameters = (Map<String, String>) map.getOrDefault("parameters", Collections.emptyMap());
+            if (parameters != null && !parameters.isEmpty()) {
+                uri.setParameters(new LinkedHashMap<>());
+                uri.getParameters().putAll(parameters);
```
Comment (apupier) [inline kamelet-support/src/main/java/io/kaoto/backend/model/deployment/kamelet/step/ToFlowStep.java:39]:
```suggestion
                uri.setParameters(new LinkedHashMap<>(parameters));
```

File: kamelet-support/src/main/java/io/kaoto/backend/model/deployment/kamelet/step/ToFlowStep.java [hunk 2]
```diff
@@ -12,23 +12,34 @@
...
+        } else if (to instanceof String sto) {
+            UriFlowStep uri = new UriFlowStep();
+            uri.setUri(sto);
+            setTo(uri);
+        } else if (to instanceof Map map) {
+            UriFlowStep uri = new UriFlowStep();
+            uri.setUri(map.getOrDefault("uri", "").toString());
+            var parameters = (Map<String, String>) map.getOrDefault("parameters", Collections.emptyMap());
+            if (parameters != null && !parameters.isEmpty()) {
+                uri.setParameters(new LinkedHashMap<>());
```
Comment (apupier) [inline kamelet-support/src/main/java/io/kaoto/backend/model/deployment/kamelet/step/ToFlowStep.java:38]: what is the interest to use a LinkedHashMap here? We want to keep an order on the elements of the Map? But given that we are retrieving them from a potentially unordered Map, it sounds like this is not ensured.

File: api/src/test/resources/io/kaoto/backend/api/resource/eip.kamelet.yaml [hunk 3]
```diff
@@ -22,6 +22,22 @@ spec:
...
+              completion-mode: MANUAL
+              steps:
+                - to:
+                    uri: direct:something
+              option:
+                - key: o1
+                  simple: ${body}
+                - key: o2
+                  expression:
+                    simple: ${body}
```
Comment (apupier) [inline api/src/test/resources/io/kaoto/backend/api/resource/eip.kamelet.yaml:40]: what do you think to add also the `sage-service`, `timeout` and `description` attributes to cover all cases with defined values?

**Comment by apupier** [comment]
current failure of jobs due to PMD violations:

```
 [INFO] PMD Failure: io.kaoto.backend.model.deployment.kamelet.step.Saga:8 Rule:UnnecessaryImport Priority:4 Unused import 'io.kaoto.backend.model.deployment.kamelet.expression.Expression'.
[INFO] PMD Failure: io.kaoto.backend.model.deployment.kamelet.step.SagaFlowStep:6 Rule:UnnecessaryImport Priority:4 Unused import 'com.fasterxml.jackson.annotation.JsonPropertyOrder'.
[INFO] PMD Failure: io.kaoto.backend.model.deployment.kamelet.step.ToFlowStep:6 Rule:UnnecessaryImport Priority:4 Unused import 'com.fasterxml.jackson.annotation.JsonPropertyOrder'.
[INFO] PMD Failure: io.kaoto.backend.model.deployment.kamelet.step.ToFlowStep:14 Rule:UnnecessaryImport Priority:4 Unused import 'java.io.Serial'.
```

File: kamelet-support/src/main/java/io/kaoto/backend/model/deployment/kamelet/step/ToFlowStep.java (hunk 2 above)
Comment (Delawen) [reply kamelet-support/src/main/java/io/kaoto/backend/model/deployment/kamelet/step/ToFlowStep.java:36]: That was a mostly unconscious thing: when debugging I noticed it was a linkedhashmap so I added it. But you are absolutely right, we have no warrantee of anything here.

File: api/src/test/resources/io/kaoto/backend/api/resource/eip.kamelet.yaml (hunk 3 above)
Comment (Delawen) [reply api/src/test/resources/io/kaoto/backend/api/resource/eip.kamelet.yaml:40]: Sure. I am just copying/pasting camel tests on this file. But we can add them.

**Pull request merged**